File: 2.Add_Two_Numbers/AddTwoNumbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* head = new ListNode(0);
        ListNode* curr = head;
        int carry = 0;
        
        while(l1 != NULL || l2 != NULL){
            int x = (l1 != NULL)? l1->val : 0;
            int y = (l2 != NULL)? l2->val : 0;
            
            int sum = x + y + carry;
            carry = sum / 10;
            curr->next = new ListNode(sum % 10);
            curr = curr->next;
            
            if(l1 != NULL)
                l1 = l1->next;
            if(l2 != NULL)
                l2 = l2->next;
        }
        if(carry > 0)
            curr->next = new ListNode(carry);
        
        return head->next;
    }
};

File: 2.Add_Two_Numbers/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        result = ListNode(0)
        temp = result
        carry = 0
        
        while l1 or l2 or carry:
            a = l1.val if l1 else 0
            b = l2.val if l2 else 0
            
            c = a + b + carry
            sm, carry = c % 10, c // 10
            
            temp.next = ListNode(sm)
            temp = temp.next
            
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        
        return result.next
